app/start_handlers.py

```diff
@@ -47,11 +47,11 @@ def setup_uac_hash(self, request, uac, lang):
     @staticmethod
     def uac_hash(uac, expected_length=16):
         if uac:
-            combined = uac.lower().replace(' ', '')
+            combined = uac.upper().replace(' ', '')
         else:
             combined = ''
 
-        uac_validation_pattern = re.compile(r'^[a-z0-9]{16}$')
+        uac_validation_pattern = re.compile(r'^[A-Z0-9]{16}$')
```

Comment:
Regex doesn't quite match [RM UAC generation](https://github.com/ONSdigital/census-rm-uac-qid-service/blob/master/src/main/resources/application-dev.yml). Not part of the reason for this change but happened to notice.

Reply:
As discussed, is OK for RHUI to have broader regex here as will always be compatible with RM view of UAC